llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -36953,6 +36934,34 @@ static void computeKnownBitsForPSADBW(SDValue LHS, SDValue RHS,
   Known = Known.zext(64);
 }
 
+static KnownBits computeKnownBitsForHorizontalOperation(
+    const SDValue Op, const APInt &DemandedElts, unsigned Depth,
+    unsigned OpIndexStart, const SelectionDAG &DAG,
+    const function_ref<KnownBits(const KnownBits &, const KnownBits &)>
+        KnownBitsFunc) {
+  APInt DemandedEltsLHS, DemandedEltsRHS;
+  getHorizDemandedEltsForFirstOperand(Op.getValueType().getSizeInBits(),
+                                      DemandedElts, DemandedEltsLHS,
+                                      DemandedEltsRHS);
+
+  const auto ComputeForSingleOpFunc =
+      [&DAG, Depth, KnownBitsFunc](const SDValue &Op, APInt &DemandedEltsOp) {
+        return KnownBitsFunc(
+            DAG.computeKnownBits(Op, DemandedEltsOp, Depth + 1),
+            DAG.computeKnownBits(Op, DemandedEltsOp << 1, Depth + 1));
+      };
+
+  if (!DemandedEltsLHS.isZero() && !DemandedEltsRHS.isZero()) {
+    return ComputeForSingleOpFunc(Op.getOperand(OpIndexStart), DemandedEltsLHS)
+        .intersectWith(ComputeForSingleOpFunc(Op.getOperand(OpIndexStart + 1),
+                                              DemandedEltsRHS));
+  }
+  if (!DemandedEltsLHS.isZero()) {
+    return ComputeForSingleOpFunc(Op.getOperand(OpIndexStart), DemandedEltsLHS);
+  }
```

Comment:
(style) remove braces from single statement

Reply:
Done. Furthermore, I've restructured the control flow a bit.